### PR TITLE
Enable Registering Submission Webhooks and Trigger IDs

### PR DESCRIPTION
**Pull request by ndharasz**
- add submission triggerId support for compute-enabled models
- add setSubmissionWebhook mutation interface

### CODE REVIEW

File: numerapi/base_api.py [hunk 1]
```diff
@@ -475,3 +475,21 @@ def get_transactions(self, model_id: str = None) -> Dict:
             utils.replace(t, "value", utils.parse_float_string)
             utils.replace(t, "insertedAt", utils.parse_datetime_string)
         return txs
+
+    def set_submission_webhook(self, model_id: str, webhook: str = None):
```
Comment (uuazed) [inline numerapi/base_api.py]: Please add a docstring and take some of the existing ones as a template.

Comment (uuazed) [inline numerapi/base_api.py]: type annotation for return type missing

File: tests/test_base_api.py [hunk 2]
```diff
@@ -149,3 +149,18 @@ def test_get_transactions(api):
         assert txs[txtype][0]["value"] == decimal.Decimal("0.4")
     assert len(txs['usdWithdrawals']) == 1
     assert txs['usdWithdrawals'][0]['usdAmount'] == decimal.Decimal("0.4")
+
+
+@responses.activate
+def test_set_submission_webhook(api):
```
Comment (uuazed) [inline tests/test_base_api.py:155]: test without an assert statement

File: numerapi/numerapi.py [hunk 3]
```diff
@@ -761,17 +761,20 @@ def upload_predictions(self, file_path: str, tournament: int = 8,
...
+                                  triggerId: $triggerId) {
                     id
                 }
             }
             '''
         arguments = {'filename': submission_auth['filename'],
                      'tournament': tournament,
-                     'modelId': model_id}
+                     'modelId': model_id,
+                     'triggerId': os.getenv('TRIGGER_ID', None)}
```
Comment (uuazed) [inline numerapi/numerapi.py:779]: Would it make sense to add a trigger_id parameter to the method signature as well?

File: numerapi/base_api.py (hunk 1 above)
Comment (uuazed) [reply numerapi/base_api.py]: I don't get what the model specific webhook is supposed to do. Is this supposed to be an URL?

**Review by uuazed — CHANGES_REQUESTED** [review]
I don't get what the trigger_id is supposed to do and what the webhook is used for. Please add some comments and address my review comments.

File: numerapi/base_api.py (hunk 1 above)
Comment (ndharasz) [reply numerapi/base_api.py]: Added this to the docstring, it's a Numerai Compute webhook https://docs.numer.ai/tournament/compute

File: numerapi/numerapi.py (hunk 3 above)
Comment (ndharasz) [reply numerapi/numerapi.py:779]: trigger_id will be set for models using numerai compute. we want a way to differentiate those using Numerai CLI to setup automated webhooks and those using a third-party service.

Comment (uuazed) [reply numerapi/numerapi.py:779]: ok, so env variable is enough here

File: numerapi/base_api.py [hunk 4]
```diff
@@ -488,8 +506,7 @@ def set_submission_webhook(self, model_id: str, webhook: str = None):
...
           }
         '''
-        arguments = {'modelId': model_id,
-                     'newSubmissionWebhook': webhook}
-        return self.raw_query(
-            query, arguments, authorization=True
-        )
+        arguments = {'modelId': model_id, 'newSubmissionWebhook': webhook}
+        res = self.raw_query(query, arguments, authorization=True)
+        print(res)
```
Comment (uuazed) [inline numerapi/base_api.py]: leftover print statement

**Pull request merged**